src/test/java/com/shapesecurity/bandolier/BundlerTest.java

```diff
@@ -142,63 +149,71 @@ public void testDeterminism() throws Exception {
 		modules.put("/js4.js", "export var d = 'd'");
 		TestLoader localLoader = new TestLoader(modules);
 
-		String actual = TestUtils.toString(TestUtils.bundleStandard("/js1.js", resolver, localLoader));
-
-		String expected = "(function(global){\n" +
-			"\"use strict\";\n" +
-			"function require(file,parentModule){\n" +
-			"if({\n" +
-			"}.hasOwnProperty.call(require.cache,file))return require.cache[file];\n" +
-			"var resolved=require.resolve(file);\n" +
-			"if(!resolved)throw new Error(\"Failed to resolve module \"+file);\n" +
-			"var module$={\n" +
-			"id:file,require:require,filename:file,exports:{\n" +
-			"},loaded:false,parent:parentModule,children:[]};\n" +
-			"if(parentModule)parentModule.children.push(module$);\n" +
-			"var dirname=file.slice(0,file.lastIndexOf(\"/\")+1);\n" +
-			"require.cache[file]=module$.exports;\n" +
-			"resolved.call(void 0,module$,module$.exports,dirname,file);\n" +
-			"module$.loaded=true;\n" +
-			"return require.cache[file]=module$.exports;\n" +
-			"}require.modules={\n" +
-			"};\n" +
-			"require.cache={\n" +
-			"};\n" +
-			"require.resolve=function(file){\n" +
-			"return{\n" +
-			"}.hasOwnProperty.call(require.modules,file)?require.modules[file]:void 0;\n" +
-			"};\n" +
-			"require.define=function(file,fn){\n" +
-			"require.modules[file]=fn;\n" +
-			"};\n" +
-			"require.define(\"1\",function(module,exports,__dirname,__filename){\n" +
-			"var __resolver=require(\"2\",module);\n" +
-			"var b=__resolver[\"b\"];\n" +
-			"var __resolver=require(\"3\",module);\n" +
-			"var c=__resolver[\"c\"];\n" +
-			"var result=\"a\"+b+c;\n" +
-			"exports[\"result\"]=result;\n" +
-			"});\n" +
-			"require.define(\"2\",function(module,exports,__dirname,__filename){\n" +
-			"var __resolver=require(\"4\",module);\n" +
-			"var d=__resolver[\"d\"];\n" +
-			"var b=\"b\"+d;\n" +
-			"exports[\"b\"]=b;\n" +
-			"});\n" +
-			"require.define(\"3\",function(module,exports,__dirname,__filename){\n" +
-			"var __resolver=require(\"4\",module);\n" +
-			"var d=__resolver[\"d\"];\n" +
-			"var c=\"c\"+d;\n" +
-			"exports[\"c\"]=c;\n" +
-			"});\n" +
-			"require.define(\"4\",function(module,exports,__dirname,__filename){\n" +
-			"var d=\"d\";\n" +
-			"exports[\"d\"]=d;\n" +
-			"});\n" +
-			"return require(\"1\");\n" +
-			"}.call(this,this).result);\n";
-
-		assertEquals(expected, actual);
+		String actualPierced = TestUtils.toString(TestUtils.bundlePierced("/js1.js", resolver, localLoader));
+		String actualStandard = TestUtils.toString(TestUtils.bundleStandard("/js1.js", resolver, localLoader));
+
+		String expectedPierced = "\"use strict\";\n" +
+				"var d=\"d\";\n" +
+				"var c=\"c\"+d;\n" +
+				"var b=\"b\"+d;\n" +
+				"var result=\"a\"+b+c;\n";
```

Comment:
`var` declarations in a module are not global. They are in scripts. As such, I believe this should all be wrapped up in an IIFE:

```js
(function(){
  "use strict";
  var d = "d";
  var c = "c" + d;
  var b = "b" + d;
  var result = "a" + b + c;
})();
```

This also removes the need for the ThisUndefiningReducer, because `this` is undefined in IIFEs with strict bodies.